spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -333,6 +352,14 @@ private Object doSendAndReceive(Object object, boolean shouldConvert) {
 					error = ((ErrorMessage) reply).getPayload();
 				}
 			}
+			if (reply == null && this.errorOnTimeout) {
```

Comment:
Thanks for both your responses to this, it looks like we were mistaken in believing the `header-enricher` was being invoked, I appreciate you pointing that out. We did a quick investigation and it looks like our automated tests in this area were not checking that our headers were being populated for the returns from these methods.

As per your suggestion it looks like our best way forward is for using the `inbound-channel-adapter` and the `<filter/>` within the `WEB-XML`.
Many thanks again for your help with this.